recipes/zlib-bytesread-to-byteswritten/src/workflow.ts

```diff
@@ -0,0 +1,96 @@
+import { getNodeRequireCalls } from "@nodejs/codemod-utils/ast-grep/require-call";
+import { getNodeImportStatements } from "@nodejs/codemod-utils/ast-grep/import-statement";
+import { resolveBindingPath } from "@nodejs/codemod-utils/ast-grep/resolve-binding-path";
+import { removeLines } from "@nodejs/codemod-utils/ast-grep/remove-lines";
+import type { Edit, Range, SgRoot } from "@codemod.com/jssg-types/main";
+import type Js from "@codemod.com/jssg-types/langs/javascript";
+
+const ZLIB_FACTORIES = [
+	"createGzip",
+	"createGunzip",
+	"createDeflate",
+	"createInflate",
+	"createBrotliCompress",
+	"createBrotliDecompress",
+	"createUnzip",
+];
+
+export default function transform(root: SgRoot<Js>): string | null {
+	const rootNode = root.root();
+	const edits: Edit[] = [];
+	const linesToRemove: Range[] = [];
+
+	// 1️ Find all zlib imports/requires
+	const nodeRequires = getNodeRequireCalls(root, "node:zlib");
+	const nodeImports = getNodeImportStatements(root, "node:zlib");
+	const importNodes = [...nodeRequires, ...nodeImports];
+	if (!importNodes.length) return null;
+
+	const factoryBindings: string[] = [];
+	const streamVariables: string[] = [];
+
+	for (const node of importNodes) {
+		const baseBind = resolveBindingPath(node, "$");
+		if (baseBind) {
+			for (const factory of ZLIB_FACTORIES) factoryBindings.push(`${baseBind}.${factory}`);
+		}
+		for (const factory of ZLIB_FACTORIES) {
+			const binding = resolveBindingPath(node, `$.${factory}`);
+			if (binding && !factoryBindings.includes(binding)) factoryBindings.push(binding);
+		}
+	}
+
+	// 2️ Track variables assigned from factories
+	for (const factory of factoryBindings) {
+		const matches = rootNode.findAll({ rule: { pattern: `const $$$VAR = ${factory}($$$ARGS)` } });
```

Comment:
Here it's only catch lexical declarations that have const. You may need a bigger query there.

Also add test for `let` & `var`